tokio-util/src/task/spawn_pinned.rs

```diff
@@ -69,7 +122,64 @@ impl LocalPoolHandle {
         Fut: Future + 'static,
         Fut::Output: Send + 'static,
     {
-        self.pool.spawn_pinned(create_task)
+        self.pool
+            .spawn_pinned(create_task, WorkerChoice::LeastBurdened)
+    }
+
+    /// Differs from `spawn_pinned` only in that you can choose a specific worker thread
+    /// of the pool, whereas `spawn_pinned` chooses the worker with the smallest
+    /// number of tasks scheduled.
+    ///
+    /// A worker thread is chosen by index. Indices are 0 based and the largest index
+    /// is given by `num_threads() - 1`
+    ///
+    /// Panics if the index is out of bounds.
+    pub fn spawn_pinned_by_idx<F, Fut>(&self, create_task: F, idx: usize) -> JoinHandle<Fut::Output>
+    where
+        F: FnOnce() -> Fut,
+        F: Send + 'static,
+        Fut: Future + 'static,
+        Fut::Output: Send + 'static,
+    {
+        self.pool
+            .spawn_pinned(create_task, WorkerChoice::ByIdx(idx))
+    }
+
+    /// Spawn a task on every worker thread in the pool and pin it so that it
+    /// can't be moved off of the thread.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::rc::Rc;
+    /// use tokio_util::task::LocalPoolHandle;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let pool = LocalPoolHandle::new(3);
+    ///
+    ///     let _ = pool.spawn_pinned_on_all_workers(|| {
+    ///         // Rc is !Send + !Sync
+    ///         let local_data = Rc::new("test");
+    ///
+    ///         // This future holds an Rc, so it is !Send
+    ///         async move { local_data.to_string() }
+    ///     });
+    /// }
+    /// ```
+    pub fn spawn_pinned_on_all_workers<F, Fut>(
+        &self,
+        create_task: F,
+    ) -> Vec<JoinHandle<Fut::Output>>
+    where
+        F: FnOnce() -> Fut,
+        F: Send + Clone + 'static,
+        Fut: Future + 'static,
+        Fut::Output: Send + 'static,
+    {
```

Comment:
I just now noticed that this is cloning the closure. I generally find that bad practice, and would prefer that this method takes an `Fn` and calls it multiple times rather than cloning it.